cloudsync/tests/test_cs.py

```diff
@@ -2992,6 +2992,50 @@ def test_cursor_tag_delete(mock_provider_generator):
     assert cs1.state.storage_get_data(cs1.emgrs[REMOTE]._cursor_tag) is None
     assert cs2.state.storage_get_data(cs2.emgrs[REMOTE]._cursor_tag) == mock_cursor_2
 
+def test_walk_tag_delete(mock_provider_generator):
+    storage_dict: Dict[Any, Any] = dict()
+    storage = MockStorage(storage_dict)
+    mock_remote_connection_id = "sharedConn"
+
+    p1 = mock_provider_generator()
+    p2 = mock_provider_generator()
+    p2.connection_id = mock_remote_connection_id
+
+    roots = ("/local", "/remote")
+
+    cs = CloudSyncMixin((p1, p2), roots, storage, sleep=None)
+    cs.do()
+
+    assert cs.state.storage_get_data(cs.emgrs[LOCAL]._walk_tag) is not None
+    assert cs.state.storage_get_data(cs.emgrs[REMOTE]._walk_tag) is not None
+
+    remote_walk_tag = cs.emgrs[REMOTE]._walk_tag
+    remote_cursor_tag = cs.emgrs[REMOTE]._cursor_tag
+
+    cs.stop()
+
+    p1 = mock_provider_generator()
+    p2 = mock_provider_generator()
+    p2.connection_id = mock_remote_connection_id
+
+    p2.disconnect()
+
+    cs1 = CloudSyncMixin((p1, p2), roots, storage, sleep=None)
+
+    # Disconnected remote provider -> can't validate root -> no walk tag
```

Comment:
a disconnected provider should forget() properly... the cs object has the root path, can just pass it in to the event forget() function right?